### PR TITLE
Feat register no machine-id

### PR DESCRIPTION
**Pull request by ahitacat**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [ ] Is this PR to correct an issue?
* [x] Is this PR an enhancement?

### Complete Description of Additions/Changes:

If insights-client is not register and there is a machine-id file in the system the registration process should fail. 
It prints a new message asking to unregister the system again.

This resolves: rhbz#2071093

### CODE REVIEW

File: insights/client/client.py [hunk 1]
```diff
@@ -126,11 +127,18 @@ def _legacy_handle_registration(config, pconn):
         write_to_disk(constants.machine_id_file, delete=True)
         logger.debug('Re-register set, forcing registration.')
 
-    logger.debug('Machine-id: %s', generate_machine_id(new=config.reregister))
+    machine_id_present = isfile(constants.machine_id_file)
 
     # check registration with API
     check = get_registration_status(config, pconn)
 
+    if machine_id_present and check['status'] is False:
```
Comment (strider) [inline insights/client/client.py:135]: Would prefer:

`if machine_id_present and not check['status']:`

File: insights/client/client.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from __future__ import print_function
 from __future__ import absolute_import
+from genericpath import isfile
```
Comment (strider) [inline insights/client/client.py:3]: Why using `genericpath.isfile` instead of `os.path.isfile`?

File: insights/client/phase/v1.py [hunk 3]
```diff
@@ -179,12 +176,15 @@ def post_update(client, config):
...
-            logger.debug('Running client in offline mode. Bypassing registration.')
-            return
-
-        if config.no_upload:
-            logger.debug("Running client without uploading. Bypassing registration.")
+        if config.offline or config.no_upload:
+            # create a machine id first thing. we'll need it for all uploads
+            logger.debug('Machine ID: %s', client.get_machine_id())
+            logger.debug("CONFIG: %s", config)
+            print_egg_versions()
```
Comment (strider) [inline insights/client/phase/v1.py:183]: Why logging all the available eggs' versions here?

File: insights/client/phase/v1.py [hunk 4]
```diff
@@ -212,23 +212,26 @@ def post_update(client, config):
...
-        return
-
-    # --payload short circuits registration check
-    if config.payload:
-        logger.debug('Uploading a specified archive. Bypassing registration.')
+    if config.offline or config.no_upload or config.payload:
+        # create a machine id first thing. we'll need it for all uploads
+        logger.debug('Machine ID: %s', client.get_machine_id())
+        logger.debug("CONFIG: %s", config)
+        print_egg_versions()
```
Comment (strider) [inline insights/client/phase/v1.py:219]: ditto

File: insights/client/phase/v1.py [hunk 5]
```diff
@@ -290,6 +298,11 @@ def post_update(client, config):
         else:
             sys.exit(constants.sig_kill_bad)
 
+    # create a machine id first thing. we'll need it for all uploads
+    logger.debug('Machine ID: %s', client.get_machine_id())
+    logger.debug("CONFIG: %s", config)
+    print_egg_versions()
```
Comment (strider) [inline insights/client/phase/v1.py:304]: ditto

**Review by strider — COMMENTED** [review]
LGTM. Only some interrogations.

**Comment by Ichimonji10** [comment]
[ESSNTL-3782](https://issues.redhat.com/browse/ESSNTL-3782)

**Comment by patriksuba** [comment]
Changes works as stated in description. While machine-id file exists client will fail with error message. Unregistering client deletes machine-id file. In BZ is also stated if the "--force" flag is also provided, print a message explaining what will happen, ask the user to confirm whether to proceed, and continue only if the user types "y" - this feature doesn't work.

**Pull request merged**